### ISSUE TITLE
New #[files] feature does not recompile on file change

### ISSUE THREAD

**Issue by GeeWee** [la10736/rstest#220]
I have a large project with many files that I read as integration tests.

I saw the new #[files] feature and love it.

_However_, I would expect it to automatically pick up when I add new files to a directory that matches the glob pattern. This does currently seem to happen. If I add a file and run the (already compiled) crate, I do not get the new test case. However if I go in and force the crate to recompile by changing some arbitrary line of code, then the new test case seems to run.

**Comment by la10736** [comment]
Sorry, isn't possible to handle this without introducing a `build.rs` script :cry: 
In this script we should define the folder that we want to handle and add a variable that should reflect the last state like an hash (but may there is already something about that in some crates).

That's really sad....

**Comment by GeeWee** [comment]
Yes I also couldn't quite figure it out how to do it in a sensible way without a `build.rs` script either :(
Might be worth documenting this though, as it took me a little while to figure out the behaviour.

**Comment by GeeWee** [comment]
It does look like something [nightly-only exists](https://github.com/rust-lang/rust/issues/73921) which might be stabilized [at some point](https://github.com/rust-lang/rust/issues/99515), not sure if we can do much except wait for that.

**Comment by la10736** [comment]
Yes... but maybe is not enough because you can track files but not folder.

**Comment by GeeWee** [comment]
Seems like it does allow for folder tracking, but it might not support globbing, which is unfortunate.

**Comment by la10736** [comment]
In #256 I wrote some notes about how to handle this in the most of the cases. Intended, with the use of `build.rs` script.

**Comment by wetneb** [comment]
Wouldn't it make sense to have a field in `Cargo.toml` to specify test resources, which would trigger a recompilation (of tests only, not of the main binary/lib) if those change?

I have opened [a discussion about this on the rust internals forum](https://internals.rust-lang.org/t/feature-request-specifying-test-resources-in-cargo-toml/22907).
